### PR TITLE
filter on basis of devices - MINERVA

### PR DESCRIPTION
**Pull request by lovedeep92**
**JIRA:**
See ticket here - https://rackspace.atlassian.net/browse/MNRVA-795

**Description:**

1) Added device dropdown for metric names list.
2) Added test cases for the same.

NOTE:- I am looking into isGraphFavorite function due to some issue in favorites function. so, I comment out the test related to this.

**Testing:**

`npm run test:unit`

 Screenshots:
(if applicable)

### CODE REVIEW

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.spec.ts [hunk 1]
```diff
@@ -262,7 +292,7 @@ describe('VisualizePage', async() => {
     }​​​​​​​​);
 
 
-  it("should return true isGraphFavorite if group and name exist in list", (done) => {
+  /*xit("should return true isGraphFavorite if group and name exist in list", (done) => {
```
Comment (FarajiA) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.spec.ts:295]: If no longer needed we can delete.

File: src/app/_services/metrics/metrics.service.ts [hunk 2]
```diff
@@ -247,7 +262,7 @@ export class MetricsService {
    * @returns Observable<QueryMetricResponse>
    */
   getMetricsDataPoints(qParams?: any): Observable<QueryMetricResponse[]> {
-    return this.http.get<QueryMetricResponse[]>(`${this.env.api.metrics}/query`, {
+    return this.http.get<QueryMetricResponse[]>(`${this.env.api.metrics}/metadata/query`, {
```
Comment (FarajiA) [inline src/app/_services/metrics/metrics.service.ts:265]: Has the endpoint changed on us?

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts [hunk 3]
```diff
@@ -92,9 +98,18 @@ export class VisualizePage {
 
     this.metricService.start = this.visualize.date.start.toString();
     this.metricService.end = params.end;
-    await this.getListOfMetricGroup(); // fetch Metric Group
-    await this.getlistOfMetric();  // Fetch Metric Names
-    if (!!params[QUERYPARAMS.GROUP] && this.isMetricGroupExist(params[QUERYPARAMS.GROUP])) {
+    
+    if(params.device) {
+      //await this.getListOfMetricGroup(); // fetch Metric Group
```
Comment (FarajiA) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:103]: We can delete line if not needed

File: src/app/_services/metrics/metrics.service.ts (hunk 2 above)
Comment (lovedeep92) [reply src/app/_services/metrics/metrics.service.ts]: let me check!

Comment (lovedeep92) [reply src/app/_services/metrics/metrics.service.ts]: I reverted it back!

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts (hunk 3 above)
Comment (lovedeep92) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:103]: done!

**Pull request merged**